src/utils/V3Utils.ts

```diff
@@ -103,6 +103,34 @@ export function isV3SlowFillLeaf<T extends MinV3SlowFillLeaf, U extends MinV2Slo
   return unsafeIsType<T, U>(slowFillLeaf, "updatedOutputAmount") && isV3RelayData(slowFillLeaf.relayData);
 }
 
+type MinV2RelayerRefundLeaf = Pick<V2RelayerRefundLeaf, "amountToReturn">;
+type MinV3RelayerRefundLeaf = Pick<V3RelayerRefundLeaf, "fillsRefundedRoot" | "fillsRefundedHash">;
+export function isV2RelayerRefundLeaf<T extends MinV2RelayerRefundLeaf, U extends MinV3RelayerRefundLeaf>(
+  leaf: T | U
+): leaf is T {
+  return unsafeIsType<T, U>(leaf, "amountToReturn") && !isV3RelayerRefundLeaf(leaf);
+}
+
+export function isV3RelayerRefundLeaf<T extends MinV3RelayerRefundLeaf, U extends MinV2RelayerRefundLeaf>(
+  leaf: T | U
+): leaf is T {
+  return unsafeIsType<T, U>(leaf, "fillsRefundedRoot");
+}
+
+type MinV2RelayerRefundExecution = Pick<V2RelayerRefundExecution, "amountToReturn">;
+type MinV3RelayerRefundExecution = Pick<V3RelayerRefundExecution, "fillsRefundedRoot" | "fillsRefundedHash">;
+export function isV2RelayerRefundExecution<T extends MinV2RelayerRefundExecution, U extends MinV3RelayerRefundExecution>(
+  leaf: T | U
+): leaf is T {
+  return unsafeIsType<T, U>(leaf, "amountToReturn") && !isV3RelayerRefundExecution(leaf);
+}
+
+export function isV3RelayerRefundExecution<T extends MinV3RelayerRefundExecution, U extends MinV2RelayerRefundExecution>(
+  leaf: T | U
+): leaf is T {
+  return unsafeIsType<T, U>(leaf, "fillsRefundedRoot");
+}
+
```

Comment:
Just flagging that I added these after merging this PR (https://github.com/across-protocol/sdk-v2/commit/0278a7d0a3addf720eee9d6cf977bd0989a0cab4), so these lines probably warrant extra scrutiny.